modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/K6ClientCodegen.java

```diff
@@ -744,9 +886,277 @@ private static String getAccept(OpenAPI openAPI, Operation operation) {
+     */
+    private Optional<OperationGrouping> extractOperationGrouping(CodegenOperation cgOperation) {
+        Optional<OperationGrouping> operationGrouping = Optional.empty();
+
+        if (cgOperation.getHasVendorExtensions() && cgOperation.vendorExtensions.containsKey(X_OPERATION_GROUPING)
+                && cgOperation.vendorExtensions.get(X_OPERATION_GROUPING) instanceof java.util.Map) {
+
+            Map.Entry<?, ?> operationGroupingEntry = ((Map<?, ?>) cgOperation.vendorExtensions
+                    .get(X_OPERATION_GROUPING)).entrySet().stream().findFirst().get();
+
+            return Optional.of(new OperationGrouping(String.valueOf(operationGroupingEntry.getKey()),
+                    Integer.parseInt(String.valueOf(operationGroupingEntry.getValue()))));
+        }
+
+        return operationGrouping;
+    }
+
+    /**
+     * If `X_OPERATION_RESPONSE` K6 vendor extension is present, check if given
+     * operation response should be hidden.
+     *
+     * @param resp
+     * @return true if should be hidden, false otherwise
+     */
+    private boolean shouldHideOperationResponse(ApiResponse resp) {
+        boolean hideOperationResponse = false;
+
+        if (Objects.nonNull(resp.getExtensions()) && !resp.getExtensions().isEmpty()
+                && resp.getExtensions().containsKey(X_OPERATION_RESPONSE)) {
+
+            Map<?, ?> respExtensions = (Map<?, ?>) resp.getExtensions().get(X_OPERATION_RESPONSE);
+
+            for (Map.Entry<?, ?> respExtensionEntry : respExtensions.entrySet()) {
+
+                if (X_OPERATION_RESPONSE_HIDE.equals(String.valueOf(respExtensionEntry.getKey()))) {
+                    hideOperationResponse = Boolean.parseBoolean(respExtensionEntry.getValue().toString());
+                }
+            }
+        }
+
+        return hideOperationResponse;
+    }
+
+    /**
+     * Check if example is present for given request body and content type.
+     *
+     * @param requestBody
+     * @param contentTypeValue
+     * @return true if present, false otherwise
+     */
+    private boolean hasRequestBodyExample(RequestBody requestBody, String contentTypeValue) {
+        return (Objects.nonNull(requestBody.getContent()) && requestBody.getContent().containsKey(contentTypeValue)
+                && Objects.nonNull(requestBody.getContent().get(contentTypeValue).getExamples())
+                && !requestBody.getContent().get(contentTypeValue).getExamples().isEmpty());
+    }
+
+    /**
+     * Extract example for given request body.
+     *
+     * @param requestBody
+     * @param contentTypeValue
+     * @param bodyOrFormParams
+     */
+    private void extractRequestBodyExample(RequestBody requestBody, String contentTypeValue,
+            List<Parameter> bodyOrFormParams) {
+
+        Optional<Map.Entry<String, Example>> requestBodyExampleEntry = requestBody.getContent().get(contentTypeValue)
+                .getExamples().entrySet().stream().findFirst();
```

Comment:
this chain `map->set->stream->optional` is quite slow, how often is this method call? might be worth to `new ArrayList(getExamples().entrySet()).get(0)`?

Reply:
Only once per request. It doesn't matter much.